### PR TITLE
Use towncrier for changelog

### PR DESCRIPTION
**Pull request by MetRonnie**

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Is this necessary for cylc-uiserver?

**Comment by MetRonnie** [comment]
The big advantage I see is we no longer have to worry about whether a changelog entry is in the right section after PRs get bumped around to different milestones

**Comment by oliver-sanders** [comment]
I get the intention, I'm just not sure the problem warrants the solution here.

**Comment by MetRonnie** [comment]
Any downside to using towncrier?

**Comment by oliver-sanders** [comment]
Extra action/integration to maintain, no obvious advantage?

**Comment by MetRonnie** [comment]
IMO never having to faff about with `CHANGES.md` is worth any minor maintenance overhead (no particular need to touch this config ever again).

File: .github/workflows/update_ui.yml [hunk 1]
```diff
@@ -55,6 +55,14 @@ jobs:
             git commit -m "ui: install ${version}"
           done
 
+      - name: Add to changelog
```
Comment (wxtim) [inline .github/workflows/update_ui.yml:58]: Nice.

**Review by wxtim — APPROVED** [review]
Looks good

**Comment by MetRonnie** [comment]
Merging with 1 review on the grounds it's working successfully on cylc-flow and the fact I had to fix another duplicate heading in `CHANGES.md`

**Pull request merged**